types/api/categories.d.ts

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+type CategorySorting = '' | 'alpha' | 'manual' | 'recent'
+type CategoryType = 'channels' | 'direct_messages' | 'favorites' | 'custom'
+
+type CategoryOrder = string[]
+type CategoryChannelOrder = string[]
+
+type CategoryChannel = {
+    id?: string;
+    user_id: string;
+    category_id: string;
+    channel_id: string;
+    sort_order: number;
+}
+
+type Category = {
+    id: string;
+    user_id: string;
+    team_id: string;
+    display_name: string;
+    sort_order: number;
+    sorting: CategorySorting;
```

Comment:
Referring to one comment above, within the Category model, you can use CategorySorting type in the description instead of giving an example.

Reply:
Same for the field `type`